images/win/scripts/ImageHelpers/VisualStudioHelpers.ps1

```diff
@@ -61,20 +61,42 @@ Function Install-VisualStudio
 }
 
 function Get-VsCatalogJsonPath {
-    $instanceFolder = Get-Item "C:\ProgramData\Microsoft\VisualStudio\Packages\_Instances\*" | Select-Object -First 1
-    return Join-Path $instanceFolder.FullName "catalog.json"
+    $instanceFolder = "C:\ProgramData\Microsoft\VisualStudio\Packages\_Instances\" + (Get-VisualStudioInstallation -VSInstallType "VS").InstanceId
+    return Join-Path $instanceFolder "catalog.json"
 }
 
-function Get-VisualStudioPath {
-    return (Get-VSSetupInstance | Select-VSSetupInstance -Product *).InstallationPath
-}
+function Get-VisualStudioInstallation {
+    Param
+    (
+        [Parameter(Mandatory)]
+        [ValidateSet('VS','BuildTools')]
+        [String] $VSInstallType
+    )
 
-function Get-VisualStudioPackages {
-    return (Get-VSSetupInstance | Select-VSSetupInstance -Product *).Packages
+    if ($VSInstallType -eq "VS")
+    {
+        $VSSelectionType = "*Enterprise*"
+    }
+    elseif ($VSInstallType -eq "BuildTools")
+    {
+        $VSSelectionType = "*Build*"
+    }
+    else
+    {
+        Write-Output "Visual Studio Installation type have to be 'VS' or 'BuildTools'"
+        exit 1
+    }
+    return Get-VSSetupInstance | Select-VSSetupInstance -Product * | Where-Object -Property DisplayName -like $VSSelectionType
 }
 
 function Get-VisualStudioComponents {
-    Get-VisualStudioPackages | Where-Object type -in 'Component', 'Workload' |
+    Param
+    (
+        [Parameter(Mandatory)]
+        [ValidateSet('VS','BuildTools')]
```

Comment:
I believe we can do `ValidateSet` only once in `Get-VisualStudioInstallation`.
It doesn't make sense to duplicate this check in every function since Get-VisualStudioInstallation will throw proper error